R/oauth-refresh.R

```diff
@@ -0,0 +1,113 @@
+# this file has its origins in oauth-refresh.R and oauth-error.R in httr
+# I want to introduce behaviour to error informatively for a deleted OAuth app
+
+# Refresh an OAuth 2.0 credential.
+#
+# Refreshes the given token, and returns a new credential with a
+# valid access_token. Based on:
+# https://developers.google.com/identity/protocols/oauth2/native-app#offline
+refresh_oauth2.0 <- function(endpoint, app, credentials, package = NULL) {
```

Comment:
Pretty faithful copy of `httr::refresh_oauth2.0()`, but with some simplifications re: things that can't happen with Google auth + a relaxation of rigid lack-of-refresh error types.